presto-spi/src/main/java/io/prestosql/spi/type/TypeUtils.java

```diff
@@ -89,6 +93,17 @@ else if (value instanceof String) {
         }
     }
 
+    public static boolean isFloatingPointNaN(Type type, Object value)
+    {
+        if (type == DOUBLE) {
```

Comment:
We should probably stop using `==` checks against types to avoid any future backward compatibility issues if we decide to create instances on the fly, and for consistency across all type checks. `.equals()` is more robust and can be implemented internally to do the `==` check as necessary.

Reply:
this is omnipresent pattern across a codebase, so we would need to (a) update all the paces and (b) have some safeguards on CI.
especially until we do (a), there isn't much of an added value in avoiding them in new code.